auth/api/iam/api.go

```diff
@@ -623,18 +617,90 @@ func (r Wrapper) oauthAuthorizationServerMetadata(ownDID *did.DID) (*oauth.Autho
 
 // OAuthClientMetadata returns the OAuth2 Client metadata for the request.Id if it is managed by this node.
 func (r Wrapper) OAuthClientMetadata(ctx context.Context, request OAuthClientMetadataRequestObject) (OAuthClientMetadataResponseObject, error) {
-	ownedDID, err := r.toOwnedDID(ctx, request.Did)
+	err := r.subjectExists(ctx, request.Subject)
 	if err != nil {
 		return nil, err
 	}
 
-	identityURL, err := createOAuth2BaseURL(*ownedDID)
+	identityURL := r.subjectToBaseURL(request.Subject)
+
+	return OAuthClientMetadata200JSONResponse(clientMetadata(identityURL)), nil
+}
+
+func (r Wrapper) OpenIDConfiguration(ctx context.Context, request OpenIDConfigurationRequestObject) (OpenIDConfigurationResponseObject, error) {
+	// find subject
+	exists, err := r.subjectManager.Exists(ctx, request.Subject)
 	if err != nil {
-		return nil, err
+		return nil, oauth.OAuth2Error{
+			Code:          oauth.ServerError,
+			Description:   "internal server error", // security reasons
+			InternalError: err,
+		}
+	}
+	if !exists {
+		return nil, oauth.OAuth2Error{
+			Code:        oauth.InvalidRequest,
+			Description: "subject not found",
+		}
+	}
+	// find DIDs for subject
+	dids, err := r.subjectManager.List(ctx, request.Subject)
```

Comment:
2nd database call, could leave out `Exists` all and check for `ErrUnknownSubject` (or something)